models/netsuite2/netsuite2__transaction_details.sql

```diff
@@ -101,19 +101,20 @@ transaction_details as (
     accounting_periods.is_closed as is_accounting_period_closed,
     accounts.name as account_name,
     accounts.type_name as account_type_name,
+    accounts.account_type_id,
     accounts.account_id as account_id,
     accounts.account_number
 
     --The below script allows for accounts table pass through columns.
     {{ fivetran_utils.persist_pass_through_columns('accounts_pass_through_columns', identifier='accounts') }},
 
     accounts.is_leftside as is_account_leftside,
-    lower(accounts.type_name) like 'accounts payable%' as is_accounts_payable,
-    lower(accounts.type_name) like 'accounts receivable%' as is_accounts_receivable,
+    lower(accounts.account_type_id) = 'acctpay' as is_accounts_payable,
+    lower(accounts.account_type_id) = 'acctrec' as is_accounts_receivable,
     lower(accounts.name) like '%intercompany%' as is_account_intercompany,
     coalesce(parent_account.name, accounts.name) as parent_account_name,
-    lower(accounts.type_name) like '%expense' as is_expense_account, -- includes deferred expense
-    lower(accounts.type_name) like '%income' as is_income_account,
+    lower(accounts.account_type_id) in ('expense', 'othexpense', 'deferexpense') as is_expense_account, -- includes deferred expense
```

Comment:
We can probably remove the comment here as it is a bit obvious with these updates.

Reply:
done